kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/ResilientKms.java

```diff
@@ -79,12 +80,14 @@ public CompletionStage<K> resolveAlias(@NonNull String alias) {
     }
 
     public <A> CompletionStage<A> retry(String name, Supplier<CompletionStage<A>> operation) {
-        return retry(name, operation, 0);
+        return retry(name, operation, 0, null);
     }
 
-    private <A> CompletionStage<A> retry(String name, Supplier<CompletionStage<A>> operation, int attempt) {
+    private <A> CompletionStage<A> retry(String name, Supplier<CompletionStage<A>> operation, int attempt, @Nullable Throwable lastFailure) {
         if (attempt >= retries) {
-            return CompletableFuture.failedFuture(new KmsException(name + " failed after " + attempt + " attempts"));
+            String lastFailureMessage = lastFailure == null ? "null" : lastFailure.getMessage();
```

Comment:
lastFailure.getMessage() might be null, so lastFailureMessage might still end up at null.
I wonder if an Option.ofNullable() might give a more succinct expression.

Reply:
much better with optional, thanks